test/harness/temporalHelpers-one-shift-time-zone.js

```diff
@@ -0,0 +1,80 @@
+// Copyright (C) 2022 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: >
+  Verify the time zone arithmetic used in TemporalHelpers.oneShiftTimeZone()
+  against known cases in the implementation's time zone database
+includes: [compareArray.js, temporalHelpers.js]
+features: [Temporal]
+---*/
+
+function checkTimeZoneArithmetic(shiftInstant, shiftNs, realTimeZoneName, shiftWallTime) {
+  // No need to test this on hosts that don't provide an Intl object. It's
+  // sufficient that the logic is tested on at least one host.
+  if (typeof globalThis.Intl === "undefined")
+    return;
```

Comment:
I'm not a big fan of this. Any suggestions on how to organize it differently are appreciated.

Reply:
Normally a test should fail if a requisite aspect is not supported, for example: https://github.com/tc39/test262/blob/main/test/built-ins/ShadowRealm/prototype/importValue/validates-realm-object.js#L10-L14